bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/util/ThingHelper.java

```diff
@@ -66,6 +66,20 @@ public static boolean equals(Thing a, Thing b) {
         if (a.getConfiguration() != null && !a.getConfiguration().equals(b.getConfiguration())) {
             return false;
         }
+        // label
+        if (a.getLabel() == null && b.getLabel() != null) {
+            return false;
+        }
+        if (a.getLabel() != null && !a.getLabel().equals(b.getLabel())) {
+            return false;
+        }
+        // location
+        if (a.getLocation() == null && b.getLocation() != null) {
+            return false;
+        }
+        if (a.getLocation() != null && !a.getLocation().equals(b.getLocation())) {
+            return false;
+        }
```

Comment:
Using Objects.equals will result in less code and is IMHO better readable.
Missed I the reason you don't use it?
https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals(java.lang.Object,%20java.lang.Object)